### PR TITLE
Add action to take time step on FD grid

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Exactly as the name says.

The action basically farms out to a system-specific implementation since if we've fallen back to FD the generic solvers just aren't robust enough. 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+class GlobalCache;
+}  // namespace Parallel
+namespace tuples {
+template <typename...>
+class TaggedTuple;
+}  // namespace tuples
+/// \endcond
+
+/// \ingroup DgSubcellGroup
+/// \brief %Actions specific to using a finite-difference subcell method.
```
Comment (fmahebert) [inline src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp]: Can the namespace documentation be moved to a more generic header?

File: tests/Unit/Evolution/DgSubcell/Actions/Test_TakeTimeStep.cpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+          ActionTesting::InitializeDataBox<initial_tags, initial_compute_tags>,
+          evolution::dg::subcell::fd::Actions::TakeTimeStep<
+              typename Metavariables::TimeDerivative>>>>;
+};
+
+template <size_t Dim>
+struct Metavariables {
+  static constexpr size_t volume_dim = Dim;
+  using component_list = tmpl::list<component<Dim, Metavariables>>;
+  // using system = System<Dim>;
```
Comment (fmahebert) [inline tests/Unit/Evolution/DgSubcell/Actions/Test_TakeTimeStep.cpp]: Delete

**Comment by nilsdeppe** [comment]
I've:
- rebased
- squashed in the removal of the commented out line
- prepended a commit that documents the namespace in `DgSubcell/Actions/Actions.hpp`

Thanks for the review :)

**Review by fmahebert — DISMISSED** [review]

File: src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+/*!
+ * \brief Take a finite-difference time step on the subcell grid.
+ *
+ * The template parameter `TimeDerivative` must have a `static apply` function
+ * that takes the `DataBox` by `gsl::not_null` as its first argument and the
+ * determinant of the inverse Jacobian from the logical to the grid frame as the
+ * second argument.
+ *
+ * GlobalCache: nothing
+ * DataBox:
```
Comment (kidder) [inline src/Evolution/DgSubcell/Actions/TakeTimeStep.hpp:46]: blank line to render correctly

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash immediately

**Comment by nilsdeppe** [comment]
I've rebased and squashed. Thanks for the reviews :)

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**